### PR TITLE
WT-8363 Redeliver Validate stats output by perf tests

### PR DESCRIPTION
**Pull request by ajmorton**
Redelivery of https://github.com/wiredtiger/wiredtiger/pull/7173 (Reverted here https://github.com/wiredtiger/wiredtiger/pull/7183).

This change is identical to the original review, but moves the `perf.send` and `s3.put` commands back into `upload-perf-test-stats` with new commit 11f6785.

### CODE REVIEW

File: bench/wtperf/wtperf_run_py/validate_expected_stats.py [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+    Example usage:
+        ./validate_perf_stats.py '{"Pages seen by eviction": 200, "Insert count": 1153326}' './evergreen_out.json'
+    """
+
+    parser = argparse.ArgumentParser()
+    parser.add_argument('expected_stats', type=str, help='The expected stats and their values')
+    parser.add_argument('stat_file', type=str, help='The evergreen stat file produced by the perf test')
+    args = parser.parse_args()
+
+    stat_file_regex = re.compile(r"evergreen_out\w*\.json")
```
Comment (Siddhartha8899) [inline bench/wtperf/wtperf_run_py/validate_expected_stats.py:50]: Maybe `[\w\-]*`?
As the file name could include hyphen: `evergreen_out_evict-fairness.json`

File: bench/wtperf/wtperf_run_py/validate_expected_stats.py [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+    if not re.match(stat_file_regex, os.path.basename(args.stat_file)):
+        print("ERROR: argument 'stat_file' should be the path to an evergreen_out*.json file")
```
Comment (Siddhartha8899) [inline bench/wtperf/wtperf_run_py/validate_expected_stats.py:52]: Printing the name of the _incorrect_ stat_file could be useful.
`print("ERROR: {} 'stat_file' should be the path to an evergreen_out*.json file".format(args.stat_file))`

**Review by Siddhartha8899 — CHANGES_REQUESTED** [review]

**Review by Siddhartha8899 — APPROVED** [review]
LGTM! mod minor changes requested.

File: bench/wtperf/wtperf_run_py/validate_expected_stats.py (hunk 1 above)
Comment (ajmorton) [reply bench/wtperf/wtperf_run_py/validate_expected_stats.py]: Updated in 0982385

File: bench/wtperf/wtperf_run_py/validate_expected_stats.py (hunk 2 above)
Comment (ajmorton) [reply bench/wtperf/wtperf_run_py/validate_expected_stats.py]: Updated in 0982385

**Review by jeremythorp — APPROVED** [review]
LGTM!

**Pull request merged**